packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+  const sameProps: FileProperty[] = givenProperties.filter((property: FileProperty) => a[property] === b[property]);
+  return sameProps.length === givenProperties.length;
+};
+
+const findDuplicateByNameAndType = (files: File[], fileToCompare: File): File | undefined =>
+  files.find((file: File) => isEqualFileBy(['name', 'type'], file, fileToCompare));
+
+type DragAndDropProps = {
+  label: string;
+  helperText: string;
+};
+
+type Language = 'en' | 'fi' | 'sv';
+
+type FileInputProps = {
+  /**
+   * The id of the input element
+   */
+  id: string;
+  /**
+   * The label for the input
+   */
+  label: string;
+  /**
+   * The label for the file button. The button is not visible for assistive technology
+   */
+  buttonLabel: string;
+  /**
+   * The language of the component. It affects which language is used to present component-specific messages, labels, and aria-labels
+   *
+   * @default "en"
+   */
+  language?: Language;
+  /**
+   * Callback fired when the list of files changes
+   */
+  onChange: (files: File[]) => void;
+  /**
+   * A comma-separated list of unique file type specifiers describing file types to allow. If present, the filename extension or filetype property is validated against the list. If the file(s) do not match the acceptance criteria, the component will not add the file(s), and it will show an error message with the file name.
+   */
+  accept?: string;
+  /**
+   * A Boolean that indicates that more than one file can be chosen
+   */
+  multiple?: boolean;
+  /**
+   * Drag and Drop area properties. If present, a drag and drop area with helper labels will render with file input. The area is not visible for assistive technology
+   */
+  dragAndDrop?: DragAndDropProps;
+  /**
+   * Maximum file size in bytes. If present, the file size is compared to this property. Maximum file size in bytes. If present, the file size is compared to this property. If the file(s) size property is larger than the max size, the component will not add the file(s), and it will show an error message with the file name.
+   */
+  maxSize?: number;
+  /**
+   * Additional class names to apply to the file input
+   */
+  className?: string;
+  /**
+   * If `true`, the file input will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * The success text content that will be shown below the input
+   */
+  successText?: string;
+  /**
+   * The error text content that will be shown below the input
+   */
+  errorText?: string;
+  /**
+   * The helper text content that will be shown below the input
+   */
+  helperText?: string;
+  /**
+   * The info text content that will be shown below the input
+   */
+  infoText?: string;
+  /**
+   * If `true`, the label is displayed as required and the `input` element will be required
+   */
+  required?: boolean;
+  /**
+   * Override or extend the styles applied to the component
+   */
+  style?: React.CSSProperties;
+};
+
+const formatBytes = (bytes: number): string => {
```

Comment:
There might be value for making separate unit tests for this format function? There seems to be some arithmetic operations and logic inside this that could be tested separately.

Reply:
Then it should be a separate utility function?

Reply:
Could we export it separately from this file, so that it could be tested?

Reply:
Yes, it might work but where would I put the tests?

Reply:
Maybe it could be tested in the same file where the FileInput component is tested? Like inside a different describe block

Reply:
- [ ] add test for formatBytes